### PR TITLE
Product/Subproduct Filter

### PR DESCRIPTION
**Pull request by amymok**
This is to add a Product/Subproduct filter

## Additions

- Added product filter

## Todos

-

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by sephcoster** [comment]
We will likely want to keep the generalized Aggregation component as part of the repository so we can use it for simpler field aggregations that only require a checkbox / expansion and take the basic parameters such as "How Complaint Submitted" or Consent values.

**Comment by JeffreyMFarley** [comment]
I think I am OK with removing the generalized `Aggregation` class even though it was a good workhorse
1.) No classes currently use it
2.) If we do need classes in the future, they will get remixed from the current components
3.) YAGNI

**Comment by sephcoster** [comment]
Agreed - I missed that "SimpleFilter" components were now handling that stuff so no reason to keep it!

File: src/Filters/Product.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    const listComponentProps = {
+      fieldName: 'product'
+    }
+
+    return (
+      <CollapsibleFilter title="Product / sub-product"
+                         desc="The type of product and sub-product the consumer identified in the complaint"
+                         showChildren={this.props.showChildren}
+                         className="aggregation">
+        <a href="http://files.consumerfinance.gov/f/documents/201704_cfpb_Summary_of_Product_and_Sub-product_Changes.pdf" target="_blank">Recent changes to products and sub-products</a>
```
Comment (sephcoster) [inline src/Filters/Product.jsx]: It looks like this aggregation needs a slightly different style and additional spacing to match the comp. That said, that would not hold up merging this but would require adding a smaller clean-up task for a revisit.

**Pull request merged**